### ISSUE TITLE
HDFS with one NameNode and no JournalNodes (no HA) for Integration Tests

### ISSUE THREAD

**Issue by fhennig** [stackabletech/hdfs-operator#266]
Druid uses HDFS for DeepStorage in its unit tests. A new HDFS instance needs to be spun up every time a test is run, and HDFS is a major contributor to the overall test duration, and also causes tests to fail occasionally because of its long startup times.

As I have learned today (thanks to Lars!), HDFS only uses a second NameNode and the JournalNodes for high availability (HA) but can also run without them. I think it would be great to have this. For an intance that only lives a few minutes and might not even see data written to it, the most bare-bones setup should be used.

**Comment by lfrancke** [comment]
It is a great idea but I'm not sure if our HDFS even supports a non-HA setup.
So we'd probably have at least one JN but only a single NN.

**Comment by maltesander** [comment]
As far as i remember:
- When creating a namenode we check if there is already an active namenode
- When creating datanodes we check that all namenode are ready or in standby mode

Ill run a quick test today.

**Comment by maltesander** [comment]
Running with nightly:
`ERROR hdfs_controller: stackable_operator::logging::controller: Failed to reconcile object controller.name="hdfsclusters.hdfs.stackable.tech" error=reconciler for object HdfsCluster.v1alpha1.hdfs.stackable.tech/simple-hdfs.default failed error.sources=[Missing node role journalnode]`

I removed the check here https://github.com/stackabletech/hdfs-operator/blob/ecacb766ae1d0937685aeab0bd3bbb22ef7064c4/rust/crd/src/lib.rs#L507
But the namenode is not coming up. So this would require more investigation.

**Comment by lfrancke** [comment]
Yes, it requires a different configuration if running without HA.
I'm not sure if we even want to support that though....

**Comment by maltesander** [comment]
Yeah but it could significantly lower the test duration / resources if we can strip it down. Or switch to S3?

**Comment by lfrancke** [comment]
We should either support it or if we don't support it then we shouldn't allow invalid configurations...

**Comment by maltesander** [comment]
Yeah, i was more talking about optimizing the tests. I think we could switch to minio for deep storage in most cases?

**Comment by fhennig** [comment]
At least for druid we could, yes.

**Comment by maltesander** [comment]
> At least for druid we could, yes.

Ok did not see this is the hdfs-operator repo...the deep storage stuff pointed me to druid.. :D
